components/map.jsx

```diff
@@ -70,6 +71,40 @@ var MarkerPopup = React.createClass({
   }
 });
 
+var ClubStatusLabel = React.createClass({
+  propTypes: {
+    status: React.PropTypes.string.isRequired,
+    showApproved: React.PropTypes.bool
+  },
+  render: function() {
+    if (this.props.status === 'pending') {
+      return (
+        <span className="label label-warning"
+         title="This club is pending approval and is not visible to other users.">
+          pending
+        </span>
+      );
+    } else if (this.props.status === 'denied') {
+      return (
+        <span className="label label-danger"
+         title="This club has been denied approval and is not visible to other users.">
+          denied
+        </span>
+      );
+    }
+    if (this.props.showApproved) {
```

Comment:
Actually, could we put this in the `else` clause above(and probably include a comment to mention status is "approved" within that clause)?  That way it's more clear that there are only 3 types of states - `pending`, `denied`, and `approved`.  

I wasn't sure if `this.props.status` could be something else and had to `console.log` to confirm.  :grin: